x-pack/plugins/secret_service/server/service/secret_service.test.ts

```diff
@@ -0,0 +1,179 @@
+    expect(unhidden.attributes).toMatchObject(hideMe);
+  });
+
+  it('should be unable to unhide values if the saved object was changed in any way', async () => {
+    expect.assertions(1);
+    const hideMe = {
+      message: 'my secret message',
+      anotherSecret: 'this is unhidden',
+      someOtherProp: 'check me',
+    };
+    let insideJob: any = null;
+    const internalSavedObject: any = null;
+    savedObjectsClient.create.mockImplementation((type: string, attributes: any, options?: any) => {
+      insideJob = {
+        id: 'testId',
+        type: 'testSecretType',
+        attributes,
+      };
+      return insideJob;
+    });
+    savedObjectsClient.get.mockImplementation((type: string, id: string, options?: any) => {
+      internalSavedObject.anotherSecret = 'I am changing the message';
+      internalSavedObject.addNewField = 'me';
+      return internalSavedObject;
+    });
+    const hidden = await subject.hideAttribute(hideMe);
+
+    try {
+      await subject.unhideAttribute(hidden);
+      fail('The test fails because no exception was thrown');
+    } catch (e) {
+      expect(e).not.toBeNull();
+    }
+  });
+
+  describe('encryption key validation', async () => {
+    it('should be valid if first time use', async () => {
+      let dummyObj: any;
+      savedObjectsClient.create.mockImplementation(
+        (type: string, attributes: any, options?: any) => {
+          dummyObj = {
+            id: options.id,
+            type,
+            attributes,
+          };
+          return dummyObj;
+        }
+      );
+      savedObjectsClient.get.mockImplementation((type: string, attributes: any, options?: any) => {
+        return dummyObj;
+      });
+      const isValid = await subject.validateKey();
+      expect(isValid).toBeTruthy();
+    });
+
+    it('should return false if decrypt fails', async () => {
+      let dummyObj: any;
+      savedObjectsClient.create.mockImplementation(
+        (type: string, attributes: any, options?: any) => {
+          attributes.secret = 'blah';
+          dummyObj = {
+            id: options.id,
+            type,
+            attributes,
+          };
+          return dummyObj;
+        }
+      );
+      savedObjectsClient.get.mockImplementation((type: string, attributes: any, options?: any) => {
+        return dummyObj;
+      });
+      const isValid = await subject.validateKey();
+      expect(isValid).toBeFalsy();
+    });
+
+    it('should return true if dummy already exists', async () => {
+      let dummyObj: any;
+      savedObjectsClient.create.mockImplementation(
+        (type: string, attributes: any, options?: any) => {
+          attributes.secret = 'blah';
+          dummyObj = {
+            id: options.id,
+            type,
+            attributes,
+          };
+          throw new Error('test error');
+        }
+      );
+      savedObjectsClient.get.mockImplementation((type: string, attributes: any, options?: any) => {
```

Comment:
__nit:__ looks like `savedObjectsClient.get` won't be called? If so, we can just use this:
```ts
savedObjectsClient.create.mockRejectedValue(new Error('test error'));
```